### PR TITLE
Proxito: don't check for index.html if the path already ends with `/`.

### PR DESCRIPTION
**Pull request by stsewd**
If the path ends with `/`, we already tried to serve the `/index.html` file and weren't able to find it, so we only need to test for the README.html in those cases, for paths that don't end with slash, we need to check for both files.

### CODE REVIEW

File: readthedocs/proxito/views/serve.py [hunk 1]
```diff
@@ -679,8 +679,15 @@ def _get_index_file_redirect(self, request, project, version, filename, full_pat
             version_type=self.version_type,
         )
 
+        tryfiles = ["index.html", "README.html"]
+        # If the path ends with `/`, we already tried to serve
+        # the `/index.html` file, so we only need to test for
+        # the `/README.html` file.
+        if full_path.endswith("/"):
+            tryfiles = ["README.html"]
```
Comment (humitos) [inline readthedocs/proxito/views/serve.py:687]: This is really good! We are reducing 1 call to S3 for _all the 404 pages_ 💪🏼

File: readthedocs/proxito/tests/test_full.py [hunk 2]
```diff
@@ -796,10 +796,42 @@ def test_404_storage_serves_custom_404_sphinx(self, storage_exists, storage_open
         self.assertEqual(response.status_code, 404)
 
     @mock.patch.object(BuildMediaFileSystemStorageTest, 'exists')
-    def test_redirects_to_correct_index(self, storage_exists):
-        """This case is when the project uses a README.html as index."""
+    def test_redirects_to_correct_index_ending_with_slash(self, storage_exists):
+        """When the path ends with a slash, we try README.html as index."""
```
Comment (humitos) [inline readthedocs/proxito/tests/test_full.py:800]: Eventually, we should remove this and stop trying `README.html`. That's a pretty legacy decision that I don't think it applies anymore.

We should probably start with new projects and keeping this behavior only for old projects and remove this logic finally after a couple of months.

**Review by humitos — APPROVED** [review]
Great you found this and changed! 👍🏼

File: readthedocs/proxito/tests/test_full.py (hunk 2 above)
Comment (davidfischer) [reply readthedocs/proxito/tests/test_full.py:800]: I'm in favor of removing it or documenting it explicitly. This is not expected default functionality from a webserver/host although it isn't a super uncommon reconfiguration.

Comment (humitos) [reply readthedocs/proxito/tests/test_full.py:800]: This is a good idea 💡

Comment (humitos) [reply readthedocs/proxito/tests/test_full.py:800]: @ericholscher we are already logging this 😉 . Take a look at this https://onenr.io/0LREdxdX5wa

There are more project than I've expected doing this 😞

Comment (stsewd) [reply readthedocs/proxito/tests/test_full.py:800]: We have 340 projects in .org  and 1 in .com https://onenr.io/0BQ1Gv2BMRx

Comment (stsewd) [reply readthedocs/proxito/tests/test_full.py:800]: we should continue this at https://github.com/readthedocs/readthedocs.org/issues/9993

**Pull request merged**